python/lsst/butler/config.py

```diff
@@ -0,0 +1,320 @@
+        config = None
+        if path.endswith('yaml'):
+            self.__initFromYamlFile(path)
+        else:
+            raise RuntimeError("Unhandled config file type:%s" % path)
+
+    def __initFromYamlFile(self, path):
+        """Opens a file at a given path and attempts to load it in from yaml.
+
+        :param path:
+        :return:
+        """
+        with open(path, 'r') as f:
+            self.__initFromYaml(f)
+
+    def __initFromYaml(self, stream):
+        """Loads a YAML config from any readable stream that contains one.
+
+        :param stream:
+        :return:
+        """
+        # will raise yaml.YAMLError if there is an error loading the file.
+        self.data = yaml.load(stream)
+        return self
+
+    def __getitem__(self, name):
+        data = self.data
+        for key in name.split('.'):
+            if data is None:
+                return None
+            if key in data:
+                data = data[key]
+            else:
+                return None
+        if isinstance(data, collections.Mapping):
+            data = Config(data)
+        return data
+
+    def __setitem__(self, name, value):
+        if isinstance(value, collections.Mapping):
+            keys = name.split('.')
+            d = {}
+            cur = d
+            for key in keys[0:-1]:
+                cur[key] = {}
+                cur = cur[key]
+            cur[keys[-1]] = value
+            self.update(d)
+        data = self.data
+        keys = name.split('.')
+        for key in keys[0:-1]:
+            data = data.setdefault(key, {})
+        data[keys[-1]] = value
+
+    def __contains__(self, key):
+        d = self.data
+        keys = key.split('.')
+        for k in keys[0:-1]:
+            if k in d:
+                d = d[k]
+            else:
+                return False
+        return keys[-1] in d
+
+    @staticmethod
+    def defaultConfigFile(productName, fileName, relativePath=None):
+        """Get the path to a default config file.
+
+        Determines a directory for the product specified by productName. Then Concatenates
+        productDir/relativePath/fileName (or productDir/fileName if relativePath is None) to find the path
+        to the default Config file
+
+        @param productName (string) The name of the product that the default config is installed as part of
+        @param fileName (string) The name of the config file. Can also include a path to the file relative to
+                                 the directory where the product is installed.
+        @param relativePath (string) The relative path from the directior where the product is installed to
+                                     the location where the file (or the path to the file) is found. If None
+                                     (default), the fileName argument is relative to the installation
+                                     directory.
+        """
+        basePath = lsst.utils.getPackageDir(productName)
```

Comment:
`getPackageDir` throws `lsst.pex.exceptions.NotFoundError` if the product can not be found so the trap on the next line is irrelevant.

Reply:
Actually just removed the whole function for now, pending discussion about default configuration locations.